core/trino-main/src/test/java/io/trino/util/DynamicFiltersTestUtil.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.util;
+
+import io.trino.spi.predicate.SortedRangeSet;
+import io.trino.spi.type.Type;
+
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+
+import static java.lang.String.format;
+
+public final class DynamicFiltersTestUtil
+{
+    private DynamicFiltersTestUtil() {}
+
+    public static String getExpectedDomainStatsString(long low, long high, int rangeCount, Type type)
```

Comment:
Could we actually generate domain and use `toString`? That would be more reliable.

Reply:
The earlier approach of using toString makes the tests pass irrespective of how the printing is done. I want tests to fail in future if the formatting changes.

Reply:
rename to `getSimplifiedDomainString` (it seems more appropriate especially when you look at tests:
```
        assertThat(domainStats.getSimplifiedDomain())
                .isEqualTo(getExpectedDomainStatsString(1L, 100L, 100, BIGINT));
```

> I want tests to fail in future if the formatting changes.

Add a comment why you want to do that

Reply:
done